File: Back/src/GuardianLog.Domain/Motorista.cs
using GuardianLog.Domain.Enum;

namespace GuardianLog.Domain;

public class Motorista : PessoaFisica
{
   public required string NumeroCNH { get; set; }
   public required string NumeroRegistroCNH { get; set; }
   public DateTime DataEmissaoCNH { get; set; }
   public DateTime DataVencimentoCNH { get; set; }
   public CategoriaCNH CategoriaCNH { get; set; }
   public TipoVinculo TipoVinculo { get; set; }
}

File: Front/GuardianLog-App/src/app/models/Motorista.ts
import { Contato } from "./Contato";
import { Endereco } from "./Endereco";

export interface Motorista {
    id: number;
    nome: string;
    sobrenome: string;
    dataNascimento: Date;
    numeroCPF: string;
    numeroRG: string;
    idOrgaoEmissor: number;
    dataEmissaoRG: Date;
    idEstadoRG: number;
    endereco: Endereco;
    idEndereco: number;
    contato: Contato;
    idContato: number;
    numeroCNH: string;
    numeroRegistroCNH: string;
    dataEmissaoCNH: Date;
    dataVencimentoCNH: Date;
    categoriaCNH: number;
    tipoVinculo: number;
}
